Swift concept methods (#2604)
Codecademy/docs: content/swift/concepts/methods/methods.md

> #Methods

Reviewer: Missing metadata! Please compare with the template:

https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md

Author: No worries. Will make the changes later or tomorrow and update the PR 🚧

Author: How's it looking? Anything else needing changes? @yangc95

Methods are functions associated with a particular type in Swift. They can be defined and called on classes, structures, and enumerations. Methods encapsulate specific tasks and functionality for working with an instance or type.

##Instance Methods

Instance methods are functions that belong to instances of a class, structure, or enumeration. They support the functionality of those instances, either by providing ways to access and modify properties or by providing functionality related to the instance's purpose.

Here's an example that defines a simple __Counter__ class, which can be used to count the number of times an action occurs:

```class Counter {
    var count = 0
    
    func increment() {
        count += 1
    }
    
    func increment(by amount: Int) {
        count += amount
    }
    
    func reset() {
        count = 0
    }
}
```

> In the above code, the __Counter__ class defines 3 instance methods:

Reviewer: Place any code snippets in backticks

1. __increment()__ increments the counter by 1
2. __increment(by: Int)__ increments the counter by a specified integer amount
3. __reset()__ resets the counter to zero

You call instance methods using dot syntax, just like accessing properties:

```let counter = Counter()
// the initial counter value is 0

counter.increment()
// counter is now 1

counter.increment(by: 5)
// counter is now 6

counter.reset()
// counter is 0 again
```

> ##The self Property

Reviewer: ```suggestion
## The `self` Property
```

---

> 
> Title: 'Methods'
> 
> Description: 'Learn about methods in Swift and how they are used to perform actions and provide functionality'
> 
> Subjects:
>     - 'iOS' 
>     - 'Mobile Development'
>     - 'Game Development'
> 
> Tags:
>     - 'Methods'
>     - 'Functions'
>     - 'Properties'
>     - 'Parameters'
> 
> CatalogContent:
> - 'Learn Swift'
> - 'Build iOS Apps with SwiftUI'
> 

Reviewer: ```suggestion
Title: 'Methods'
Description: 'Learn about methods in Swift and how they are used to perform actions and provide functionality'
Subjects:
  - 'iOS' 
  - 'Mobile Development'
  - 'Game Development'
Tags:
  - 'Methods'
  - 'Functions'
  - 'Properties'
  - 'Parameters'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
```

---

> Title: 'Swift Methods'

Reviewer: ```suggestion
Title: 'Methods'
```

> Description: 'Learn about methods in Swift and how they are used to perform actions and provide functionality

Reviewer: Typo here, forgot to close this string with a tick

Can you also match this to the beginning definition in the intro paragraph below?

Subjects:
    - 'iOS' 
    - 'Mobile Development'
    - 'Game Development'

Tags:
    - 'Methods'
    - 'Functions'
    - 'Properties'
    - 'Parameters'

CatalogContent:
- 'Learn Swift'
- 'Build iOS Apps with SwiftUI'

---

> ## Methods

Reviewer: ```suggestion
```

> In Swift, methods are like actions that an object (belonging to a class) can perform. Imagine a class as a blueprint for creating objects, and methods are the functions that these objects can execute. For example, think of a "Car" class that has a method called "startEngine." When creating a "Car" object, use the "startEngine" method to make the car's engine come to life.

Reviewer: not just classes, can we make that edit. Also, it's great that there is an example here, though can this be two separate paragraphs? Explain methods in a abstract way and then go into examples - better transition for the reader this way

Reviewer: @Wakodono classes, structs, and enums can have instance methods - can we add this change and also link out to the entries:

- https://www.codecademy.com/resources/docs/swift/classes
- https://www.codecademy.com/resources/docs/swift/structures
- https://www.codecademy.com/resources/docs/swift/enums

> ## Syntax of Methods
> 
> ### To define a method in Swift, follow these steps:

Reviewer: ```suggestion
## Creating Methods
```

1. Use the `func` keyword, which stands for "function."
2. Name the method.
3. Add a pair of parentheses `()` after the method name.
4. If the method takes input (parameters), list them within the parentheses and provide their data types.

> 5. Place the method's code within a set of curly braces `{}`.

Reviewer: Turn list into paragraph format

```swift
class Car {
    var brand = ""
    var color = ""

    // Method to start the car's engine
    func startEngine() {
        print("Starting the engine of the \(brand) car.")
    }
}
```

> In the code above, there is a class called "Car" with two properties: `brand` and `color`. As well as a method named `startEngine`, which does not take any parameters. When this method is called on a "Car" object, it will print a message about starting the car's engine.

Reviewer: Good! Make sure to place any code mentions in backticks

> ## Creating Objects and Using Methods

Reviewer: ```suggestion
## Using Methods
```

To utilize a method, an instance of the class must be created. This instance will possess the characteristics specified by the class, allowing access to its methods.

```swift
var myCar = Car()
myCar.brand = "Codecar"
myCar.color = "Shadow Grey"

myCar.startEngine()
// Output: Starting the engine of the Codecar car.
```

> In the example above, we created an object `myCar` of the class "Car" and set its `brand` to "Codecar" and `color` to "Shadow Grey." Then, we called the `startEngine()` method on the `myCar` object, which printed the message about starting the car's engine.

Reviewer: Rephrase without pronouns like "we" (style related)

## Conclusion

> Methods in Swift allow us to define behaviours for our objects and make them more interactive. They provide a way for objects to perform specific actions and enhance the functionality of our classes. By using methods, we can make our Swift code more organized, reusable, and efficient.

Reviewer: ```suggestion
Methods in Swift allow us to define behaviors for our objects and make them more interactive. They provide a way for objects to perform specific actions and enhance the functionality of our classes. By using methods, we can make our Swift code more organized, reusable, and efficient.
```

Reviewer: Can this be moved in the intro section? Remove the header here
